clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriter.java

```diff
@@ -158,10 +169,20 @@ public void open(Map<String, String> checkpointedInfo, Optional<Supplier<byte[]>
       if (lastFinishedSSTFileNo < 0) {
         throw new VeniceException("Last finished sst file no: " + lastFinishedSSTFileNo + " shouldn't be negative");
       }
-      makeSureAllPreviousSSTFilesBeforeCheckpointingExist();
-      removeSSTFilesAfterCheckpointing();
-      currentSSTFileNo = lastFinishedSSTFileNo + 1;
+      if (doesAllPreviousSSTFilesBeforeCheckpointingExist()) {
+        // remove the unwanted sst files, as flow will continue from the checkpointed info
+        removeSSTFilesAfterCheckpointing(this.lastFinishedSSTFileNo);
+        currentSSTFileNo = lastFinishedSSTFileNo + 1;
+      } else {
+        // remove all the temp sst files if found any as we will start fresh
+        removeSSTFilesAfterCheckpointing(-1);
```

Comment:
Can we replace `-1` with a constant, e.g. `REMOVE_ALL_SST_FILES` or whatever?

Reply:
Done, thanks.